src/display/process/allProcs.go

```diff
@@ -146,37 +155,54 @@ func AllProcVisuals(dataChannel chan []*proc.Process,
 				ui.Close()
 				wg.Done()
 				return
+			case "?":
+				helpVisible = !helpVisible
 			case "<Resize>":
 				updateUI() // updateUI only during resize event
-			case "s": //s to pause
-				pause()
-			case "j", "<Down>":
-				myPage.BodyList.ScrollDown()
-			case "k", "<Up>":
-				myPage.BodyList.ScrollUp()
-			case "<C-d>":
-				myPage.BodyList.ScrollHalfPageDown()
-			case "<C-u>":
-				myPage.BodyList.ScrollHalfPageUp()
-			case "<C-f>":
-				myPage.BodyList.ScrollPageDown()
-			case "<C-b>":
-				myPage.BodyList.ScrollPageUp()
-			case "g":
-				if previousKey == "g" {
+			}
+			if helpVisible {
+				switch e.ID {
+				case "?":
+					updateUI()
+				case "<Escape>":
+					helpVisible = false
+					updateUI()
+				}
+				ui.Render(help)
+			} else {
+				switch e.ID {
+				case "?":
+					updateUI()
+				case "s": //s to pause
+					pause()
+				case "j", "<Down>":
+					myPage.BodyList.ScrollDown()
+				case "k", "<Up>":
+					myPage.BodyList.ScrollUp()
+				case "<C-d>":
+					myPage.BodyList.ScrollHalfPageDown()
+				case "<C-u>":
+					myPage.BodyList.ScrollHalfPageUp()
+				case "<C-f>":
+					myPage.BodyList.ScrollPageDown()
+				case "<C-b>":
+					myPage.BodyList.ScrollPageUp()
+				case "g":
+					if previousKey == "g" {
+						myPage.BodyList.ScrollTop()
+					}
+				case "<Home>":
 					myPage.BodyList.ScrollTop()
+				case "G", "<End>":
+					myPage.BodyList.ScrollBottom()
 				}
-			case "<Home>":
-				myPage.BodyList.ScrollTop()
-			case "G", "<End>":
-				myPage.BodyList.ScrollBottom()
-			}
 
-			ui.Render(myPage.Grid)
-			if previousKey == "g" {
-				previousKey = ""
-			} else {
-				previousKey = e.ID
+				ui.Render(myPage.Grid)
+				if previousKey == "g" {
+					previousKey = ""
+				} else {
+					previousKey = e.ID
+				}
 			}
```

Comment:
This entire switch is a little messy, it's what's causing the non scrolling issue for the help widget. Maybe boil this down to An if condition and two switch cases? 

```go
if helpVisible {
  // handle keyboard input for the help bar.
  // switch case for e.ID
} else {
  // handle keyboard input for main menu.
  // switch case for e.ID
}
```
EDIT: This can be done for all the other pages too I feel.

Reply:
This is pretty much what's happening right now. The switch is inside the if condition to handle each individual key

Reply:
@anihm136 right now there are 3 different switches. They go like: 
```go 
case e := ui.Events:
  switch e.ID {
    // This handles close and resize
  }
  if helpVisible {
    // cases for help widget
  } else {
    // cases for main UI
  }

Reply:
This can be condensed is all.